charts/core/src/YAxis/YAxis.tsx

```diff
@@ -96,6 +96,22 @@ export function YAxis({ type, label, unit }: YAxisProps) {
 
   useEffect(() => {
     updateChartOptions(getOptions({ type, label, unit, theme }));
+
+    return () => {
+      /**
+       * Hides the axis when the component is unmounted.
```

Comment:
Is there a cleaner way to handle unmount?

Reply:
I'm not sure. Definitely open to ideas! The problem is as noted in [this comment](https://github.com/mongodb/leafygreen-ui/pull/2529#discussion_r1823225539). So the goal here is to essentially manually trigger a chart config update any time the component unmounts.

I'll actually chuck this out to the team async and see if anyone has any cleaner ideas for this!